public/pages/Findings/containers/Findings/Findings.tsx

```diff
@@ -157,44 +158,57 @@ class Findings extends Component<FindingsProps, FindingsState> {
     this.setState({ loading: true });
     const { findingsService, detectorService, notifications } = this.props;
     try {
-      const detectorsRes = await detectorService.getDetectors();
-      if (detectorsRes.ok) {
-        const detectors = detectorsRes.response.hits.hits;
+      // Get all the Findings logic
+      const findingRes = await findingsService.getFindings();
+
+      if (findingRes.ok) {
         const ruleIds = new Set<string>();
         let findings: FindingItemType[] = [];
 
-        const detectorId = this.props.match.params['detectorId'];
-        for (let detector of detectors) {
-          if (!detectorId || detector._id === detectorId) {
-            const findingRes = await findingsService.getFindings({ detectorId: detector._id });
-
-            if (findingRes.ok) {
-              const detectorFindings: FindingItemType[] = findingRes.response.findings.map(
-                (finding) => {
-                  finding.queries.forEach((rule) => ruleIds.add(rule.id));
-                  return {
-                    ...finding,
-                    detectorName: detector._source.name,
-                    logType: detector._source.detector_type,
-                    detector: detector,
-                  };
-                }
-              );
-              findings = findings.concat(detectorFindings);
-            } else {
-              errorNotificationToast(notifications, 'retrieve', 'findings', findingRes.error);
+        // Assuming getDetectors returns an array of detectors
+        const detectorsRes = await detectorService.getDetectors();
+        if (detectorsRes.ok) {
+          const detectors = detectorsRes.response.hits.hits;
+          const detectorInfoMap = new Map();
+
+          detectors.forEach((detector) => {
+            detectorInfoMap.set(detector._id, {
+              logType: detector._source.detector_type,
+              name: detector._source.name,
+              detector: detector,
+            });
+          });
+          const detectorFindingsPromises = findingRes.response.findings.map((finding) => {
+            try {
+              // Use detectorInfoMap to get detector information
+              const detectorInfo = detectorInfoMap.get(finding.detectorId);
+              finding.queries.forEach((rule) => ruleIds.add(rule.id));
+              return {
```

Comment:
Let's simplify this to set the detector related info before the return so we have a single check for `detectorInfo`

```
const detectorData = detectorInfo ? {
  detectorName: detectorInfo.name
  ...
} : {
  detectorName: DEFAULT....
}
```